packages/actor-rdf-metadata-extract-void/lib/ActorRdfMetadataExtractVoid.ts

```diff
@@ -0,0 +1,547 @@
+import type { ActorInitQueryBase } from '@comunica/actor-init-query';
+import { QueryEngineBase } from '@comunica/actor-init-query';
+import type {
+  IActionRdfMetadataExtract,
+  IActorRdfMetadataExtractOutput,
+  IActorRdfMetadataExtractArgs,
+} from '@comunica/bus-rdf-metadata-extract';
+import { ActorRdfMetadataExtract } from '@comunica/bus-rdf-metadata-extract';
+import type { IActorTest, TestResult } from '@comunica/core';
+import { passTestVoid } from '@comunica/core';
+import type * as RDF from '@rdfjs/types';
+import { LRUCache } from 'lru-cache';
+import { RdfStore } from 'rdf-stores';
+import { termToString } from 'rdf-string-ttl';
+import type { Algebra } from 'sparqlalgebrajs';
+
+/**
+ * A comunica Void RDF Metadata Extract Actor.
+ */
+export class ActorRdfMetadataExtractVoid extends ActorRdfMetadataExtract {
+  private readonly queryEngine: QueryEngineBase;
+  private readonly inferUriRegexPattern: boolean;
+  private readonly bindingsCache: LRUCache<string, RDF.Bindings[]>;
+
+  public static readonly VOID = 'http://rdfs.org/ns/void#';
+  public static readonly SPARQLSD = 'http://www.w3.org/ns/sparql-service-description#';
+
+  public constructor(args: IActorRdfMetadataExtractVoidArgs) {
+    super(args);
+    this.queryEngine = new QueryEngineBase(args.actorInitQuery);
+    this.inferUriRegexPattern = args.inferUriRegexPattern;
+    this.bindingsCache = new LRUCache<string, RDF.Bindings[]>({ max: args.bindingsCacheSize });
```

Comment:
Is this cache really needed?
If really so, then we need to listen to cache invalidation.

Reply:
It is probably not strictly needed, but I figured it would make sense, since a lot of the formulae end up pulling the same properties from the VoID descriptions for different purposes (multipliers, divisors, etc.). I think I will remove this cache thing, and if performance becomes an issue later, it can be added back.